service/controller/resource/heartbeat/create.go

```diff
@@ -16,35 +17,34 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 	r.logger.LogCtx(ctx, "level", "debug", "message", "checking if heartbeat exists")
 	var current heartbeat.Heartbeat
 	getResult, err := r.heartbeatClient.Get(ctx, desired.Name)
+
 	if IsApiNotFoundError(err) {
 		r.logger.LogCtx(ctx, "level", "debug", "message", "heartbeat does not exist")
 		r.logger.LogCtx(ctx, "level", "debug", "message", "creating heartbeat")
-		req := &heartbeat.AddRequest{
-			Name:          desired.Name,
-			Description:   desired.Description,
-			Interval:      desired.Interval,
-			IntervalUnit:  heartbeat.Unit(desired.IntervalUnit),
-			Enabled:       &desired.Enabled,
-			OwnerTeam:     desired.OwnerTeam,
-			AlertMessage:  desired.AlertMessage,
-			AlertTag:      desired.AlertTags,
-			AlertPriority: desired.AlertPriority,
+		err := r.createHeartbeat(ctx, desired)
+		if err != nil {
+			return microerror.Mask(err)
 		}
-		addResult, err := r.heartbeatClient.Add(ctx, req)
+		r.logger.LogCtx(ctx, "level", "debug", "message", "created heartbeat")
+
+		// The initial ping to the heartbeat is there to move the heartbeat from inactive to active.
+		_, err = r.heartbeatClient.Ping(ctx, desired.Name)
 		if err != nil {
 			return microerror.Mask(err)
 		}
 
-		r.logger.LogCtx(ctx, "level", "debug", "message", "created heartbeat")
-		current = addResult.Heartbeat
+		return nil
 	} else if err != nil {
 		return microerror.Mask(err)
-	} else {
-		r.logger.LogCtx(ctx, "level", "debug", "message", "heartbeat exists")
-		current = getResult.Heartbeat
 	}
 
+	r.logger.LogCtx(ctx, "level", "debug", "message", "heartbeat exists")
+	current = getResult.Heartbeat
+
 	r.logger.LogCtx(ctx, "level", "debug", "message", "checking if heartbeat needs update")
+	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("Current: %v", current))
```

Comment:
Do we need this log line, and the one after?

Reply:
No, I am using them to debug the heartbeats getting updated